server/core/src/main/java/io/whitefox/core/services/AwsGlueConfigBuilder.java

```diff
@@ -0,0 +1,33 @@
+package io.whitefox.core.services;
+
+import io.whitefox.core.AwsCredentials;
+import io.whitefox.core.MetastoreProperties;
+import io.whitefox.core.aws.utils.StaticCredentialsProvider;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.aws.AwsClientProperties;
+import org.apache.iceberg.aws.AwsProperties;
+
+public class AwsGlueConfigBuilder {
+
+  public Map<String, String> buildConfig(
+      MetastoreProperties.GlueMetastoreProperties glueMetastoreProperties) {
+    AwsCredentials.SimpleAwsCredentials credentials =
```

Comment:
Check the cast before performing it and throw a descriptive error in case the cast is not possible. Otherwise we would get a ClassCastException which is much harder to debug